api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -272,6 +272,14 @@ message JwtProvider {
   //
   string header_in_metadata = 14;
 
+  // If true, clear the route cache in order to allow this filter to impact routing decisions.
```

Comment:
Does this always happen, or only when some material change occurs in the filter?

Reply:
Always happens semantic is more stable in the long because we don't have to worry about a `default_metadata`, or `on_failure_metadata`.

Thus `always happens` is an attribute of the provider rather than an condition of the provider's execution result.

@qiwzhang @lizan may have a better sense of the guideline

Reply:
I am trying to implement and I realize there is grouped provider: Any, All, and even allow missing

Additional clear route post action need to defined on the group.

for simplicity we can assume the eventual clear action is an Any semantic. More specifically,

1. requires_any: [a, b, c]
if *any* of a,b,c sub requirements need clear route, `require_any` clear route.

2. requires_and: [a, b, c]
if *any* of a,b,c sub requirements need clear route, `require_and` clear route.

This might introduce false clear, but it's much simpler, not depends on the execution.
After all, it's still much better than using another filter or blindly clearing route

CC @qiwzhang @yangminzhu

Reply:
Does the above clear_route_action on grouping case make sense?

Reply:
I feel this is a bit complicated and just not sure if it worth the complexity.

In Istio, we just simply always clear the route cache in the `istio_authn` filter when we do JWT validation. We do not distinguish whether it's from an individual JWT provider or not, so a simple global flag is very simple and good enough.

Second, we also use the `allow_missing` field. It is triggered when a request has no JWT token. In this case, we still want to clear the route cache in case other filters has written some data to the dynamic metadata that might affect the routing action. (this should be rare since I think other filters should generally not do it in the first place).

Last, my natural thought to the `clear_route_action` in each JWT provider is that it only clears the cache when the corresponding jwt provider is triggered. The require_all, require_any list could include multiple JWT providers and/or even the `allow_missing` provider, it's not very clear of the semantics when only a subset of the providers are setting the flag.